README.md
---
cover: .gitbook/assets/pyeuy7iyfw961.png
coverY: -32.8928046989721
---

# Patates Sevenler Discord Sunucusunun Dokümanları

[Copyright © 2021 Patates Sevenler](https://www.dmca.com/Protection/Status.aspx?ID=043db8b8-347d-4caa-ba02-f3d351094105\&refurl=https://patatessevenler.com/)

{% hint style="warning" %}
* Bu dokümantasyon sayfasıdır, asıl site değildir.
* Asıl siteye [patatessevenler.com](https://patatessevenler.com/)[ ](https://patatessevenler.com/)adresinden gidebilirsiniz.
* Discord'da [📬│destek](https://discord.com/channels/560140367362523136/851308144180199464)[ ](https://discord.com/channels/560140367362523136/851308144180199464)kanalı üzerinden canlı destek alabilirsiniz.
{% endhint %}

![ Patates Gibi Her Halinizle Güzel Olun](.gitbook/assets/806719130233208832.png)

{% embed url="https://www.youtube.com/watch?v=02Amt4HyiLc" %}
Patates Sevenler Tanıtım
{% endembed %}

![](.gitbook/assets/standard.gif)

## Bilgilendirme

> Burası Patates Sevenlerin yapım aşamasında olan dokümantasyon sayfasıdır. Aradığınız bilgiyi bulamazsanız, lütfen [📬│destek](https://discord.com/channels/560140367362523136/851308144180199464) kanalından bizleri bilgilendirin.

## Sorun Giderme

> Eğer sunucuda herhangi bir sorun tespit ederseniz, yetkililere bildirmeden önce aşağıdaki sorunlar sayfasından zaten bilinen bir sorun olup olmadığını kontrol ediniz.

{% content-ref url="diger/sorunlar.md" %}
[sorunlar.md](diger/sorunlar.md)
{% endcontent-ref %}

## Öneride Bulunmak

> Yeni bir özellik veya iyileştirme öneriniz varsa, [🤯│öneri ](https://discord.com/channels/560140367362523136/813034318913470505)kanalına önerinizi yazarak tüm üyelerin değerlendirmesine sunabilirsiniz.&#x20;

## İletişim

* _**Sosyal Medya**_ <img src="https://cdn.discordapp.com/emojis/829699064534663219.png?v=1" alt=":insta:" data-size="line">

{% embed url="http://www.instagram.com/patates_sevenler" %}

* **E-mail**

$$
admin@patatessevenler.com
$$


changelog.md
---
description: >-
  Buradan sunucuda yapılan yenilikleri, değişiklikleri ve hata düzeltmelerini
  öğrenebilirsiniz.
---

# Changelog

{% hint style="danger" %}
Dükkanı kapattık mrq!

* No longer supported!
{% endhint %}

## 12/10/2021

### Yenilikler

* 🌟 PPLİK GİFLER 🌟, 💈 PARTNERLİK 💈 ve 📰 HABERLER 📰 kategorileri arsive kaldirildilar.
* 🤐Tanışıksız Patates rolunun tum yaptirimlari kaldirildi.
* K\*rt patates rolu 🌾Çiftçi Patates olarak yeniden isimlendirildi
* Gay patates rolu 🎩Boy Patates olarak yeniden isimlendirildi.
* &#x20;Giriste random olarka atilan hos geldin mesaji sadece hos geldiin yazicak sekilde degistirildi.

## 25/9/2021

### Yenilikler

* Hapishanede olan banli uyeler sunucudan ciktiklarinda otomatik olarak ipden banlanacaklar.

## 27/8/2021

### Hata Düzeltmeleri

* **🔮│öneren\_patatesler** kanalina link atilamama hatasi giderildi.
* **🔢sayı\_say kanalina** **📏Sözelci Patates** rolu sebebiyle mesaj atilamama hatasi giderildi.

## 26/8/2021

### Yenilikler

* **!dilen** komudunun **maksimum** coin odulu **500** cikartildi.
* **🔢sayı\_say** kanalinda sayilan her sayi icin **10** coin odul eklendi.
* **🏆│sıralama** adında yeni bir kanal eklendi.
* **💥│spamcı\_patatesler** kanali deneysel olarak aktif edildi.
* **🔮│öneren\_patatesler** kanali istek uzerine yeniden aktif edildi.
* **✨promosyon** kanali arsivden cikartilarak yeni ozellikleriyle sunuldu. [https://forms.gle/JVUKWgRWDbsiXQuN8](https://forms.gle/JVUKWgRWDbsiXQuN8)
* Admin panele **✨promosyon\_başvuru** adinda yeni bir kanal eklendi.
* **📣yetkili\_duyuru** kanalina her discord coktugunde atilan embed artik @here etiketini etikeletleyecek.
* **/top** ve **/rank** komutlarinin kanal sinirlamalari kaldirildi.
* **!cuvalim** komudunun thumbnaili degistirildi.
* **!replo** komudu **!topcoin** olarak degistirildi.
* **!leader** komudu **!topkayit** olarak degistirildi.
* **!topsayim** adinda sayim siralamasini gosteren yeni bir komut eklendi.
* Dokumantasyona **🏆Level Komutları** adinda yeni bir komut rehberi eklendi.
* **🔢sayı\_say** kanalinda kullanilan CC degistirildi ve dbler yeni ccye aktarildi;
* Artik ust uste sayi sayilabilecek.
* Sayim levelleri **100**, **500** ve **1000** olmak uzere degistirildi.
* Sayim levelleri bot tarafindan otomatik olarak verilecek.

### Hata Düzeltmeleri

* Siralama embedlerinde bos yere sebep olan sunucudan cikan uyelerin verileri veri tabanlarindan silindi.
* **🔇mod\_act** kanali kullanilmadigi halde yer kaplamasi sorunu kanali arsivliyerek cozumlendi.
* **🔍┋görünümler** kanalindaki embedleri yonetem tepki komudu yeniden elden gecirilerek yetkililerin satiln alabilmesi gereken kanallari satin alamamalari hatasi ve herhangi bir gorunum rolune sahipseniz diger embedlerden tepkinizi cektiginizde embedin kanali gizlediniz yazma hatasi giderildi.

## 24/7/2021

### Yenilikler

* **📚patates\_coin\_bilgi** kanalindaki rehber emojiler eklenerek degistirilmistir.
* **!cuvalim** komudu **!cuval** olarak degistidildi.
* **!coinekle** komudu **!odul** olarak degistirildi.
* **!coinsil** komudu **!ceza** olarak degistirildi.
* **!coinat** komudu **!transfer** olarak degistirildi.
* **!slot** kumar komudu artik loglanabiliyor.
* Bazi ekonomi komutlarinin tasarimlari degistirildi.

## 20/7/2021

### Yenilikler

* **💣Booster Patates** rolunden audit logu gorebilme yetkisi alindi.
* **🍖 Kemik Patates** rolune audit logu gorebilme yetkisi verildi.
* **🎂DOĞUM GÜNÜ Patatesi**, **💎Haftanın Patatesleri** ve **🎑Zıbab Patates** rollerine herkes tarafindan etiketlenilebilme izni verildi.

### Hata Düzeltmeleri

* **🎖️Old Patates** rolunun everyone cekebilmesi hatasi giderildi.
* **👩‍🍳Hamarat Patates**, **🧐Nezih Patates** ve **Rank** rollerinin herkes tarafindan etiketlenilebilmesi hatasi giderildi.
* Yetkili rollerinin etiketlenilememesi hatasi giderildi.



## 18/7/2021

### Yenilikler

* Sesli kanallarin hepsinin isminde ve sablonunda degisiklikler yapildi.
* Boosterlar icin olan **│📝log** ve **│🔑hesap\_yaş** kanallari arsive kaldirildi.
* Nazar ayraci bazi android cihazlarda gozukmedigi icin tekerlekli sandalyede oturan adam ayracina degistirildi.

### Hata Düzeltmeleri

* Birden fazla renk rolune sahip uyelerin renk rolleri temizlendi.
* Birden fazla rank rolune sahip uyelerin rank rolleri temizlendi.

## 18/7/2021

### Yenilikler

* **🔧┋özel\_ayar**, **🔨┋booster\_ayar** ve **🔩┋özel\_ayraç** kanallari arsive kaldirildi.
* **⏤⏤⏤| 🛒 MAĞAZA 🛒|⏤⏤⏤** adli yeni bir kategori acildi ve **🌈┋renkler** **🍭┋özel\_renk 🔗┋ayraçlar 🔍┋özel\_görünüm** **🧰┋ıvır\_zıvır** kanallari bu kategoriye tasindi.
* **🔗┋ayraçlar** kanalinin icerigi degistirildi.
* Sunucuya yeni ayrac secenekleri eklendi ve artik coin ile satin alinabilecek.
* **🧰┋ıvır\_zıvır** adinda yeni bir kanal eklendi, bu kanal sayesinde uyeler belli basli ozellikler satin alabilecekler.
* **🔍┋özel\_görünüm** ve **🔍┋özel\_görünüm** kanallari artik **💣Booster Patates** ve **🍖 Kemik Patates** rollerine sahip uyelere tamamiyle gorunur kilindi.

### Hata Düzeltmeleri

* **🙋│kendini\_tanıt** kanalinin etkinlik rolu alindiginda gizlenmesine sebep olan hata giderildi.
* Sekreterlerin bazi istatistik kanallarini gorememe hatasi giderildi.
* **!kimsin** komudunda istatistik emojisi yeniden calisir hale getirildi.
* 1 den fazla ayrac grubuna sahip uyelerin ayraclari temizlendi.
* Sesli kanallarda level 4 isimli eski arsivlenmis bir rolun izinleri kaldirildi.

## 15/7/2021

### Yenilikler

* Discordun cokmesi durumunda otomatik yetkilileri etiketleyen bir feed olusturuldu.
* **📻Radyocu Bot** rolu silindi ve bu role sahip botlar **🎵Müzikci Bot** kategorisine tasindi.
* **📊| İstatistik |📊** kategorisi altina sunucudaki **kanal** ve **rol** sayilarini sayan istatistik kanallari eklendi.

### Hata Düzeltmeleri

* **🤖| Bot Durumları |🤖** kategorisi altindaki kanallarin aciklamalarindaki yanlis bot etiketleri duzeltildi.

## 13/7/2021

### Yenilikler

* 7 ana rengin renk kodlari degistirildi.

### Hata Düzeltmeleri


Dashboard/dc_login.py
"""
Example of using Discord OAuth to allow someone to
log in to your site. The scope of 'email+identify' only
lets you see their email address and basic user id.
"""
from requests_oauthlib import OAuth2Session
import getpass
from flask import Flask, request, redirect, session, render_template
import os
import pymongo
from pymongo import MongoClient

# Disable SSL requirement
os.environ['OAUTHLIB_INSECURE_TRANSPORT'] = '1'

# Settings for your app
base_discord_api_url = 'https://discordapp.com/api'
client_id = r'' # Get from https://discordapp.com/developers/applications
client_secret = ""
redirect_uri='http://localhost:8080/callback'
scope = ['identify', 'guilds']
token_url = 'https://discordapp.com/api/oauth2/token'
authorize_url = 'https://discordapp.com/api/oauth2/authorize'
cluster = MongoClient("")
db = cluster["discordveri"]
collection = db["uyeveri"]
project_root = os.path.dirname(os.path.realpath('__file__'))
template_path = os.path.join(project_root, 'app/templates')
static_path = os.path.join(project_root, 'app/static')
app = Flask(__name__, template_folder=template_path, static_folder=static_path)
app.secret_key = os.urandom(24)

@app.route("/")
def home():
    """
    Presents the 'Login with Discord' link
    """
    oauth = OAuth2Session(client_id, redirect_uri=redirect_uri, scope=scope)
    login_url, state = oauth.authorization_url(authorize_url)
    session['state'] = state
    print("Login url: %s" % login_url)
    return '<a href="' + login_url + '">Login with Discord</a>'
    application = app

@app.route("/callback")
def oauth_callback():
    """
    The callback we specified in our app.
    Processes the code given to us by Discord and sends it back
    to Discord requesting a temporary access token so we can 
    make requests on behalf (as if we were) the user.
    e.g. https://discordapp.com/api/users/@me
    The token is stored in a session variable, so it can
    be reused across separate web requests.
    """
    discord = OAuth2Session(client_id, redirect_uri=redirect_uri, state=session['state'], scope=scope)
    token = discord.fetch_token(
        token_url,
        client_secret=client_secret,
        authorization_response=request.url,
    )
    session['discord_token'] = token


    discord2 = OAuth2Session(client_id, token=session['discord_token'])
    response = discord2.get(base_discord_api_url + '/users/@me')
    # https://discordapp.com/developers/docs/resources/user#user-object-user-structure
    print('Profile: %s' % response.json()['id'])
    response2 = discord2.get(base_discord_api_url + '/users/@me/guilds')
    print('Guilds: %s' % response2.json())
    post = {"_id": response.json()['id'], "sunucular": response2.json()}
    collection.insert_one(post)


    return render_template('index.html')

application = app





Dashboard/dc_logintest.py
from requests_oauthlib import OAuth2Session
import getpass
from flask import Flask, request, redirect, session, render_template
import os
import pymongo
from pymongo import MongoClient

# Disable SSL requirement
os.environ['OAUTHLIB_INSECURE_TRANSPORT'] = '1'

# Settings for your app
base_discord_api_url = 'https://discordapp.com/api'
client_id = r'' # Get from https://discordapp.com/developers/applications
client_secret = ""
redirect_uri='http://localhost:8080/callback'
scope = ['identify', 'guilds']
token_url = 'https://discordapp.com/api/oauth2/token'
authorize_url = 'https://discordapp.com/api/oauth2/authorize'
cluster = MongoClient("")
db = cluster["discordveri"]
collection = db["uyeveri"]
db2 = cluster["myFirstDatabase"]
collection2 = db2["memberstats"]

app = Flask(__name__)
app.secret_key = os.urandom(24)

@app.route("/")
def home():
    oauth = OAuth2Session(client_id, redirect_uri=redirect_uri, scope=scope)
    login_url, state = oauth.authorization_url(authorize_url)
    session['state'] = state
    print("Login url: %s" % login_url)
    return render_template('login.html', login= login_url)

@app.route("/callback", methods=["GET", "POST"])
def oauth_callback():
    discord = OAuth2Session(client_id, redirect_uri=redirect_uri, state=session['state'], scope=scope)
    token = discord.fetch_token(
        token_url,
        client_secret=client_secret,
        authorization_response=request.url,
    )
    session['discord_token'] = token


    discord2 = OAuth2Session(client_id, token=session['discord_token'])
    response = discord2.get(base_discord_api_url + '/users/@me')
    # https://discordapp.com/developers/docs/resources/user#user-object-user-structure
    id = response.json()['id']
    session['id'] = id
    avatar = response.json()['avatar']
    session['avatar'] = avatar
    session['username'] = response.json()['username']
    response2 = discord2.get(base_discord_api_url + '/users/@me/guilds')
    collection.delete_one({"uyeid": id})
    post = {"uyeid": response.json()['id'],"username": response.json()['username'], "sunucular": response2.json()}
    collection.insert_one(post)
    return redirect("/dash")

@app.route("/dash", methods=["GET", "POST"])
def dash():
    id = session['id']
    avatar = session['avatar']

    results = collection2.find({"userID": id})
    for result in results:
        try:
            mesaj = result["totalChatStats"]
        except:
            mesaj = "0"

    results = collection2.find({"userID": id})
    for x in results:
        try:
            haftamesaj = x["chatStats"]['592651806362959895']
        except:
            haftamesaj = "0"

    
    results = collection2.find({"userID": id})
    for result in results:
        try:
            millis = result["totalVoiceStats"]
            millis = int(millis)
            seconds=(millis/1000)%60
            seconds = int(seconds)
            minutes=(millis/(1000*60))%60
            minutes = int(minutes)
            hours=(millis/(1000*60*60))
            hours = int(hours)
        except:
            hours = "0"
            minutes = "0"
            seconds = "0"
        
    
    results = collection2.find({"userID": id})
    for x in results:
        try:
            milli2 = x["voiceStats"]['592651874528919564']
            milli2 = int(milli2)
            second2=(milli2/1000)%60 
            second2 = int(second2)
            minute2=(milli2/(1000*60))%60
            minute2 = int(minute2)
            hour2=(milli2/(1000*60*60))
            hour2 = int(hour2)
        except:
            hour2 = "0"
            minute2 = "0"
            second2 = "0"

    return render_template('index.html', username= session['username'], avatar= f'https://cdn.discordapp.com/avatars/{id}/{avatar}.png', hours= hours, minutes= minutes, mesaj= mesaj, hour2= hour2, minute2= minute2, haftamesaj= haftamesaj)



# Or run like this
# FLASK_APP=discord_oauth_login_server.py flask run -h 0.0.0.0 -p 8000
if __name__ == '__main__':
    app.run(host='localhost', port=8080)



Dashboard/discord_oauth_login_server.py
"""
Example of using Discord OAuth to allow someone to
log in to your site. The scope of 'email+identify' only
lets you see their email address and basic user id.
"""
from requests_oauthlib import OAuth2Session
import getpass
from flask import Flask, request, redirect, session, render_template
import os
import pymongo
from pymongo import MongoClient

# Disable SSL requirement
os.environ['OAUTHLIB_INSECURE_TRANSPORT'] = '1'

# Settings for your app
base_discord_api_url = 'https://discordapp.com/api'
client_id = r'' # Get from https://discordapp.com/developers/applications
client_secret = ""
redirect_uri='http://localhost:8080/callback'
scope = ['identify', 'guilds']
token_url = 'https://discordapp.com/api/oauth2/token'
authorize_url = 'https://discordapp.com/api/oauth2/authorize'
cluster = MongoClient("")
db = cluster["discordveri"]
collection = db["uyeveri"]

app = Flask(__name__)
app.secret_key = os.urandom(24)

@app.route("/")
def home():
    """
    Presents the 'Login with Discord' link
    """
    oauth = OAuth2Session(client_id, redirect_uri=redirect_uri, scope=scope)
    login_url, state = oauth.authorization_url(authorize_url)
    session['state'] = state
    print("Login url: %s" % login_url)
    return '<a href="' + login_url + '">Login with Discord</a>'

@app.route("/callback")
def oauth_callback():
    """
    The callback we specified in our app.
    Processes the code given to us by Discord and sends it back
    to Discord requesting a temporary access token so we can 
    make requests on behalf (as if we were) the user.
    e.g. https://discordapp.com/api/users/@me
    The token is stored in a session variable, so it can
    be reused across separate web requests.
    """
    discord = OAuth2Session(client_id, redirect_uri=redirect_uri, state=session['state'], scope=scope)
    token = discord.fetch_token(
        token_url,
        client_secret=client_secret,
        authorization_response=request.url,
    )
    session['discord_token'] = token


    discord2 = OAuth2Session(client_id, token=session['discord_token'])
    response = discord2.get(base_discord_api_url + '/users/@me')
    # https://discordapp.com/developers/docs/resources/user#user-object-user-structure
    print('Profile: %s' % response.json()['id'])
    response2 = discord2.get(base_discord_api_url + '/users/@me/guilds')
    print('Guilds: %s' % response2.json())
    post = {"_id": response.json()['id'], "sunucular": response2.json()}
    collection.insert_one(post)


    return render_template('index.html')



# Or run like this
# FLASK_APP=discord_oauth_login_server.py flask run -h 0.0.0.0 -p 8000
if __name__ == '__main__':
    app.run(host='localhost', port=8080)



Dashboard/passenger_wsgi.py
from requests_oauthlib import OAuth2Session
import getpass
from flask import Flask, request, redirect, session, render_template
import os
import pymongo
from pymongo import MongoClient

# Disable SSL requirement
os.environ['OAUTHLIB_INSECURE_TRANSPORT'] = '1'

base_discord_api_url = 'https://discordapp.com/api'
client_id = r'' # Get from https://discordapp.com/developers/applications
client_secret = ""
redirect_uri='https://dash.patatessevenler.com/callback'
scope = ['identify', 'guilds']
token_url = 'https://discordapp.com/api/oauth2/token'
authorize_url = 'https://discordapp.com/api/oauth2/authorize'
cluster = MongoClient("")
db = cluster["discordveri"]
collection = db["uyeveri"]
db2 = cluster["myFirstDatabase"]
collection2 = db2["memberstats"]
project_root = os.path.dirname(os.path.realpath('__file__'))
template_path = os.path.join(project_root, 'templates')
static_path = os.path.join(project_root, 'static')
app = Flask(__name__, template_folder=template_path, static_folder=static_path)
app.secret_key = os.urandom(24)

@app.route("/")
def home():
    oauth = OAuth2Session(client_id, redirect_uri=redirect_uri, scope=scope)
    login_url, state = oauth.authorization_url(authorize_url)
    session['state'] = state
    return render_template('login.html', login= login_url)

@app.route("/callback", methods=["GET", "POST"])
def oauth_callback():
    discord = OAuth2Session(client_id, redirect_uri=redirect_uri, state=session['state'], scope=scope)
    token = discord.fetch_token(
        token_url,
        client_secret=client_secret,
        authorization_response=request.url,
    )
    session['discord_token'] = token


    discord2 = OAuth2Session(client_id, token=session['discord_token'])
    response = discord2.get(base_discord_api_url + '/users/@me')
    # https://discordapp.com/developers/docs/resources/user#user-object-user-structure
    id = response.json()['id']
    session['id'] = id
    avatar = response.json()['avatar']
    session['avatar'] = avatar
    session['username'] = response.json()['username']
    response2 = discord2.get(base_discord_api_url + '/users/@me/guilds')
    collection.delete_one({"uyeid": id})
    post = {"uyeid": response.json()['id'],"username": response.json()['username'], "sunucular": response2.json()}
    collection.insert_one(post)
    return redirect("/dash")

@app.route("/dash", methods=["GET", "POST"])
def dash():
    id = session['id']
    avatar = session['avatar']

    results = collection2.find({"userID": id})
    for result in results:
        try:
            mesaj = result["totalChatStats"]
        except:
            mesaj = "0"

    results = collection2.find({"userID": id})
    for x in results:
        try:
            haftamesaj = x["chatStats"]['592651806362959895']
        except:
            haftamesaj = "0"

    
    results = collection2.find({"userID": id})
    for result in results:
        try:
            millis = result["totalVoiceStats"]
            millis = int(millis)
            seconds=(millis/1000)%60
            seconds = int(seconds)
            minutes=(millis/(1000*60))%60
            minutes = int(minutes)
            hours=(millis/(1000*60*60))
            hours = int(hours)
        except:
            hours = "0"
            minutes = "0"
            seconds = "0"
        
    
    results = collection2.find({"userID": id})
    for x in results:
        try:
            milli2 = x["voiceStats"]['592651874528919564']
            milli2 = int(milli2)
            second2=(milli2/1000)%60 
            second2 = int(second2)
            minute2=(milli2/(1000*60))%60
            minute2 = int(minute2)
            hour2=(milli2/(1000*60*60))
            hour2 = int(hour2)
        except:
            hour2 = "0"
            minute2 = "0"
            second2 = "0"
        
    return render_template('index.html', username= session['username'], avatar= f'https://cdn.discordapp.com/avatars/{id}/{avatar}.png', hours= hours, minutes= minutes, mesaj= mesaj, hour2= hour2, minute2= minute2, haftamesaj= haftamesaj)

application = app



